### PR TITLE
[DI][Config] Make accessing environment variables in configuration mo…

### PR DESCRIPTION
**Pull request by nicolas-grekas**
https://github.com/symfony/symfony/pull/18155

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/ExpressionLanguageProvider.php [hunk 1]
```diff
@@ -38,6 +39,20 @@ public function getFunctions()
...
+
+            new ExpressionFunction('env', function ($arg, $default = null) {
+                if (2 > func_num_args()) {
+                    return sprintf('$this->getEnvironmentVariable(%s)', $arg);
+                }
+
+                return sprintf('$this->getEnvironmentVariable(%s, %s)', $arg, $default);
+            }, function (array $variables, $value, $default = null) {
+                if (2 > func_num_args()) {
+                    return $variables['container']->getEnvironmentVariable($value);
```
Comment (magnusnordlander) [inline src/Symfony/Component/DependencyInjection/ExpressionLanguageProvider.php:51]: Unless I am quite mistaken, this (and the same three lines down) won't work with `getEnvironmentVariable` declared as protected. I would suggest keeping it public.

**Comment by magnusnordlander** [comment]
If the above is adressed, I'm fine with these changes.

File: src/Symfony/Component/DependencyInjection/ExpressionLanguageProvider.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/ExpressionLanguageProvider.php:51]: Indeed, fixed!

**Comment by nicolas-grekas** [comment]
If you're OK @magnusnordlander , please make this patch yours.
The command for fetching it with git is:
`git fetch https://github.com/nicolas-grekas/symfony.git refs/pull/3/head`
Then, we'd need a few tests of course.

BTW, of did you feel about this way of helping (opening a PR on my fork with an alternative patch then giving back it as suggested above?) For me, it looks more efficient than commenting sometimes, should I continue with other PRs in your opinion?

**Comment by magnusnordlander** [comment]
Cool. I've pushed it to my PR.

Regarding the way of helping, I'm quite git savvy, so I didn't have any trouble pulling in your changes, but I might imagine that does not apply to everyone. I think what's probably easier (and what I've done previously) is to open a PR against the original PR branch. That way the contributor can merge your changes via the Github merge button. The changeset will of course get a merge commit, but from what I understand the Symfony Github toolset rebases every commit automatically anyway, so it'll still come out as a single commit in the end, once the original PR is merged.

**Comment by nicolas-grekas** [comment]
Cleaning the git history is also a way to help... But I'll provide a few more git commands to help with this you're right!

**Pull request closed without merging**